hc_dna/src/zome/capabilities.rs

```diff
@@ -0,0 +1,149 @@
+/*!
+hc_dna::zome::capabilities is a set of structs for working with holochain dna.
+*/
+
+extern crate serde_json;
+
+use wasm::DnaWasm;
+
+/// Enum for Zome Capability "membrane" property.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub enum Membrane {
+    #[serde(rename = "public")]
+    Public,
+    #[serde(rename = "agent")]
+    Agent,
+    #[serde(rename = "api-key")]
+    ApiKey,
+    #[serde(rename = "zome")]
+    Zome,
+}
+
+impl Default for Membrane {
+    /// Default zome capability membrane is "public"
+    fn default() -> Self {
+        Membrane::Public
+    }
+}
+
+/// Represents the "capability" sub-object on a "zome" "capabilities" object.
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq)]
+pub struct CapabilityType {
+    /// How visibility should be handled for this capability.
+    #[serde(default)]
+    pub membrane: Membrane,
+}
+
+impl Default for CapabilityType {
+    /// Defaults for a "capability" sub-object on a "zome" "capabilities" object.
+    fn default() -> Self {
+        CapabilityType {
+            membrane: Membrane::Public,
```

Comment:
Yeah, in this case, since all the struct members themselves have the defaults we want, we could use derive. I mostly went this way for consistency with the other dna structs, and in case we want to change something in the future, it will require less refactoring.